chia/wallet/wallet_nft_store.py

```diff
@@ -110,5 +139,25 @@ async def get_nft_by_id(self, nft_id: bytes32) -> Optional[NFTCoinInfo]:
             None if row[2] is None else LineageProof.from_json_dict(json.loads(row[2])),
             Program.from_bytes(row[5]),
             uint32(row[3]),
+            uint32(row[6]),
             row[4] == IN_TRANSACTION_STATUS,
         )
+
+    async def rollback_to_block(self, height: int) -> None:
+        """
+        Rolls back the blockchain to block_index. All coins confirmed after this point are removed.
+        All coins spent after this point are set to unspent. Can be -1 (rollback all)
+        """
+        if height < 0:
+            return
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            # Remove reorged NFTs
+            await (await conn.execute("DELETE FROM users_nfts WHERE latest_height>?", (height,))).close()
```

Comment:
shouldn't you also remove all the tx records for this block height?

Reply:
It's handled in the tx store. It doens't delete the tx. It just revert the spent and confirmed flags.